core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -309,10 +312,11 @@ protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startin
    * @param dataFilter a data filter
    * @param dataFiles data files to validate have no new row deletes
    * @param caseSensitive whether expression binding should be case-sensitive
+   * @param ignoreEqualityDeletes whether equality deletes should be ignored in validation
    */
   protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startingSnapshotId,
                                                   Expression dataFilter, Iterable<DataFile> dataFiles,
-                                                  boolean caseSensitive) {
+                                                  boolean caseSensitive, boolean ignoreEqualityDeletes) {
```

Comment:
I think this makes sense for the internal API, but I generally prefer new method names for booleans, like `validateNoNewPositionDeletesForDataFiles` and `validateNoNewDeletesForDataFiles`.

Reply:
Maybe we should add some comment to indicate why we could ignore the equality deletes and when we should ignore or not.

Reply:
Added a new method and moved this to a private method, please let me know if that is enough